### ISSUE TITLE
Read Once, Write Once

### ISSUE THREAD

**Issue by shakna-israel** [shakna-israel/docbuilder#81]
Rather than streaming input in, opening and closing before each line to try and accommodate for race conditions, simply bite the bullet and preload files.

Load the entire file into a list, and close it. Parse it from memory and build the entire output into memory, and then write it.

Race conditions can be mentioned, like 

```
if file hash changed:
    print("Input file may have changed... Continue?")
```

**Comment by shakna-israel** [comment]
This obviously increases the memory docbuilder may use exponentially, but it should benefit from a speed increase.

**Comment by shakna-israel** [comment]
Solved by a4a43b8fb02cdd217b539861649b2d953f7dfbd7.
